gluten-core/src/main/scala/org/apache/spark/sql/execution/datasources/GlutenWriterColumnarRules.scala

```diff
@@ -80,60 +77,80 @@ case class FakeRowAdaptor(child: SparkPlan) extends UnaryExecNode with GlutenPla
+        "org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat".equals(
+          command.table.storage.outputFormat.get)
+      case _: CreateHiveTableAsSelectCommand =>
+        throw new UnsupportedOperationException(
+          "native writer cannot recognize command: " + cmd.getClass.getName +
+            " please append `using parquet` in your CTAS query")
+      case _ =>
+        throw new UnsupportedOperationException(
+          "native writer cannot recognize command: " + cmd.getClass.getName)
     }
   }
 
   case class NativeWritePostRule(session: SparkSession) extends Rule[SparkPlan] {
     override def apply(p: SparkPlan): SparkPlan = p match {
-      case rc @ DataWritingCommandExec(cmd, child) if isGlutenInsertInto(cmd) =>
-        child match {
-          // if the child is columnar, we can just wrap&transfer the columnar data
-          case c2r: ColumnarToRowExecBase =>
-            rc.withNewChildren(Array(FakeRowAdaptor(c2r.child)))
-          // If the child is aqe, we make aqe "support columnar",
-          // then aqe itself will guarantee to generate columnar outputs.
-          // So FakeRowAdaptor will always consumes columnar data,
-          // thus taking no risk to downgrade performance
-          case aqe: AdaptiveSparkPlanExec =>
-            rc.withNewChildren(
-              Array(
-                FakeRowAdaptor(
-                  AdaptiveSparkPlanExec(
-                    aqe.inputPlan,
-                    aqe.context,
-                    aqe.preprocessingRules,
-                    aqe.isSubquery,
-                    supportsColumnar = true
-                  ))))
-          case other => rc.withNewChildren(Array(FakeRowAdaptor(other)))
+      case rc @ DataWritingCommandExec(cmd, child) =>
+        session.sparkContext.setLocalProperty("isNativeParquetAppliable", "false")
+        if (isNativeParquetAppliable(cmd)) {
+          session.sparkContext.setLocalProperty("isNativeParquetAppliable", "true")
+          child match {
+            // if the child is columnar, we can just wrap&transfer the columnar data
+            case c2r: ColumnarToRowExecBase =>
+              rc.withNewChildren(Array(FakeRowAdaptor(c2r.child)))
+            // If the child is aqe, we make aqe "support columnar",
+            // then aqe itself will guarantee to generate columnar outputs.
+            // So FakeRowAdaptor will always consumes columnar data,
+            // thus avoiding the case of c2r->aqe->r2c->writer
+            case aqe: AdaptiveSparkPlanExec =>
+              rc.withNewChildren(
+                Array(
+                  FakeRowAdaptor(
+                    AdaptiveSparkPlanExec(
+                      aqe.inputPlan,
+                      aqe.context,
+                      aqe.preprocessingRules,
+                      aqe.isSubquery,
+                      supportsColumnar = true
+                    ))))
+            case other => rc.withNewChildren(Array(FakeRowAdaptor(other)))
```

Comment:
It is better to check `ColumnarToRowExec` then add `FakeRowAdaptor`, as the output of child may be `Row`, not `ColumnarBatch`